probe/shell/shell.go

```diff
@@ -66,12 +65,8 @@ func (s *Shell) DoProbe() (bool, string) {
 	ctx, cancel := context.WithTimeout(context.Background(), s.ProbeTimeout)
 	defer cancel()
 
-	for _, e := range s.Env {
-		v := strings.Split(e, "=")
-		os.Setenv(v[0], v[1])
-	}
-
 	cmd := exec.CommandContext(ctx, s.Command, s.Args...)
+	cmd.Env = append(os.Environ(), s.Env...)
```

Comment:
This gets the environment of the easeprobe executable and appends our new Env (status,name,rtt etc)? Is so this is not advisable. Just set the new environment variables and let the system inherit/propagate the environment variables based on their export criteria...

Reply:
I did some tests and read the source code.
for Go StdLib,
- if we define the `cmd.Env`, it only sets those env variables, the parent's or system env won't be set.
- if the `cmd.Env` is not defined. then use the system default env, which is the `os.Environ()`

So, if we need to pass the system env to the child process, we have to do `cmd.Env := append(os.Environ(), env…)`

I think the current implementation is right.

Reply:
It is a clear and thorough analysis.👍

Reply:
> I did some tests and read the source code. for Go StdLib,
> 
> * if we define the `cmd.Env`, it only sets those env variables, the parent's or system env won't be set.
> * if the `cmd.Env` is not defined. then use the system default env, which is the `os.Environ()`
> 
> So, if we need to pass the system env to the child process, we have to do `cmd.Env := append(os.Environ(), env…)`
> 
> I think the current implementation is right.

It is a clear and thorough analysis.👍 @haoel